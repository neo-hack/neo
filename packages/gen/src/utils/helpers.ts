import { Workflow } from '../interface'
import { hooks } from '../utils/hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: () => {
            ob.complete()
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
