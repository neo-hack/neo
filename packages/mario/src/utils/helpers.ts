import { Workflow } from '../interface'
import { hooks } from './hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: (args: { error: Error }) => {
            if (!args.error) {
              ob.complete()
              return
            }
            ob.error(args.error)
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
