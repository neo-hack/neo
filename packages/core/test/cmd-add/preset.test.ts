import { clearLockFile, execNeo, readLockFile, storeDir } from '../helpers'

afterEach(() => {
  clearLockFile()
})

describe('load preset', () => {
  it('load preset from npm package should work', async () => {
    await execNeo(['add', '@aiou/preset-aiou', '--store-dir', storeDir, '--preset'])
    expect(readLockFile()).toMatchSnapshot()
  }, 10000)
})
