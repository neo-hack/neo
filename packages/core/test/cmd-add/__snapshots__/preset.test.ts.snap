// Vitest Snapshot v1

exports[`load preset > load preset from npm package should work 1`] = `
{
  "presets": {
    "@aiou/preset-aiou": {
      "configs": [
        {
          "name": "PNPM CI workflow",
          "pref": "./assets/workflows/pnpm/ci.yaml",
        },
        {
          "name": "PNPM Release workflow",
          "pref": "./assets/workflows/pnpm/release.yaml",
        },
        {
          "name": "PNPM Snapshot Release workflow",
          "pref": "./assets/workflows/pnpm/snapshot-release.yaml",
        },
        {
          "name": "Rust CI workflow",
          "pref": "./assets/workflows/rust/ci.yaml",
        },
        {
          "name": "Rust Homebrew Release workflow",
          "pref": "./assets/workflows/rust/release.homebrew.yaml",
        },
        {
          "name": "SWC plugin Release workflow",
          "pref": "./assets/workflows/rust/release.swc.yaml",
        },
        {
          "name": "SWC plugin Snapshot-Release workflow",
          "pref": "./assets/workflows/rust/snapshot-release.swc.yaml",
        },
      ],
      "templates": [
        {
          "name": "@aiou/ts-lib-template",
        },
        {
          "name": "@aiou/actions-template",
        },
        {
          "name": "@aiou/rollup-plugin-template",
        },
        {
          "name": "@aiou/rollup-template",
        },
        {
          "name": "@aiou/dayjs-plugin-template",
        },
        {
          "name": "@aiou/bin-template",
        },
        {
          "name": "@aiou/webext-template",
        },
        {
          "name": "@aiou/eslint-config-template",
        },
        {
          "name": "@aiou/webpack-loader-template",
        },
        {
          "name": "@aiou/react-template",
        },
        {
          "name": "@aiou/react-components-lib-template",
        },
        {
          "name": "@aiou/original-html-template",
        },
        {
          "name": "@aiou/chrome-extenstion-template",
        },
      ],
    },
  },
  "version": undefined,
}
`;
